src/facade/dragonfly_connection.cc

```diff
@@ -316,6 +314,36 @@ QueueBackpressure& GetQueueBackpressure() {
 
 thread_local vector<Connection::PipelineMessagePtr> Connection::pipeline_req_pool_;
 
+class PipelineCacheSizePaceMaker {
```

Comment:
nit: maybe PipelineWatermarkTracker

Reply:
how about `PipelineCacheSizeTracker` ? (not strongly opnionated so let me know which one you prefer!)